src/reducers/activeRequests.js

```diff
@@ -1,18 +1,25 @@
-import { Set } from 'immutable'
+// @flow
 
-import { actionReducer } from './utils'
+import produce from 'immer'
 import { ADD_ACTIVE_REQUEST, REMOVE_ACTIVE_REQUEST } from '_/constants'
+import { actionReducer } from './utils'
+
+type ActiveRequestsStateType = Array<string>
 
-const initialState = Set()
+const initialState: ActiveRequestsStateType = []
 
 const activeRequestsReducer = actionReducer(initialState, {
-  [ADD_ACTIVE_REQUEST] (state, { payload: requestId }) {
-    return state.add(requestId)
-  },
+  [ADD_ACTIVE_REQUEST]: produce((draft: ActiveRequestsStateType, { payload: requestId }) => {
+    const flatRequestId: string = JSON.stringify(requestId)
```

Comment:
we do we need to stringify the ID?

Reply:
To make sure `draft.includes(flatRequestId)` and `draft.filter(request => request !== flatRequestId)` work no matter what the requestId type is.

Reply:
I think we can skip stringify - previously we relied on the reference equality and we can continue doing so. However some comment explaining that would be helpful.
Check:
```
const a = { a: 'a' }
const set = Set().add(a)
const arr = [a]

arr.includes({ a: 'a' }) == set.has({ a: 'a' }) == false
arr.includes(a) == set.has(a) == true
```

Reply:
> I think we can skip stringify - previously we relied on the reference equality and we can continue doing so.

So I tried it and realized that relying on `Object.is()` or `===` object ref check will NOT work in this context.  I think it comes down to Immer doing its magic.  The object stored in the produced array is not the same as the array going in.

And of course, it is a pitfill: https://immerjs.github.io/immer/pitfalls#drafts-arent-referentially-equal

Reply:
OK .But still using stringfy is confusing. Actually naming an object of this type `requestId` is confusing!
How about  destructuring: ` { payload: {method, url})` and later just `draft.find(r => r.url == url && r.method == method)` ?